src/Security/MemberAuthenticator/ChangePasswordHandler.php

```diff
@@ -63,102 +66,152 @@ public function __construct($link, MemberAuthenticator $authenticator)
             $message = DBField::create_field(
                 'HTMLFragment',
                 '<p>' . _t(
-                    'SilverStripe\\Security\\Security.ENTERNEWPASSWORD',
+                    Security::class . '.ENTERNEWPASSWORD',
                     'Please enter a new password.'
                 ) . '</p>'
             );
 
-            // Subsequent request after the "first load with hash" (see previous if clause).
             return [
                 'Content' => $message,
-                'Form'    => $this->changePasswordForm()
+                'Form' => $this->changePasswordForm()
             ];
         }
 
+        // Logged in user requested a password change form.
         if (Security::getCurrentUser()) {
-            // Logged in user requested a password change form.
             $message = DBField::create_field(
                 'HTMLFragment',
                 '<p>' . _t(
-                    'SilverStripe\\Security\\Security.CHANGEPASSWORDBELOW',
+                    Security::class . '.CHANGEPASSWORDBELOW',
                     'You can change your password below.'
                 ) . '</p>'
             );
 
             return [
                 'Content' => $message,
-                'Form'    => $this->changePasswordForm()
-            ];
-        }
-        // Show a friendly message saying the login token has expired
-        if ($token !== null && $member && !$member->validateAutoLoginToken($token)) {
-            $message = DBField::create_field(
-                'HTMLFragment',
-                _t(
-                    'SilverStripe\\Security\\Security.NOTERESETLINKINVALID',
-                    '<p>The password reset link is invalid or expired.</p>'
-                    . '<p>You can request a new one <a href="{link1}">here</a> or change your password after'
-                    . ' you <a href="{link2}">log in</a>.</p>',
-                    [
-                        'link1' => Security::lost_password_url(),
-                        'link2' => Security::login_url(),
-                    ]
-                )
-            );
-
-            return [
-                'Content' => $message,
+                'Form' => $this->changePasswordForm()
             ];
         }
 
         // Someone attempted to go to changepassword without token or being logged in
         return Security::permissionFailure(
             Controller::curr(),
             _t(
-                'SilverStripe\\Security\\Security.ERRORPASSWORDPERMISSION',
+                Security::class . '.ERRORPASSWORDPERMISSION',
                 'You must be logged in in order to change your password!'
             )
         );
     }
 
-
     /**
-     * @param Member $member
-     * @param string $token
+     * Set the encrypted session token for the member.
      */
-    protected function setSessionToken($member, $token)
+    protected function setSessionToken(Member $member, string $token, bool $alreadyEncrypted = false): void
```

Comment:
I've opted to keep the existing method name with the same default behaviour it has in CMS 5 so we don't have to deprecate it.